src/devices/SocketCan/Interop.cs

```diff
@@ -0,0 +1,232 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Net.Sockets;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Serialization;
+using System.Text;
+
+namespace Iot.Device.SocketCan
+{
+    internal class Interop
+    {
+        private const int PF_CAN = 29;
+
+        // SFF = Standard Frame Format - 11 bit
+        public const uint CAN_SFF_MASK = 0x000007FF;
+
+        // EFF = Extended Frame Format - 29 bit
+        public const uint CAN_EFF_MASK = 0x1FFFFFFF;
+        public const uint CAN_ERR_MASK = 0x1FFFFFFF;
+
+        public const int SOL_CAN_BASE = 100;
+        public const int SOL_CAN_RAW = SOL_CAN_BASE + (int)CanProtocol.CAN_RAW;
+
+
+        [DllImport("libc", EntryPoint = "socket", CallingConvention = CallingConvention.Cdecl)]
+        private static extern int CreateNativeSocket(int domain, int type, CanProtocol protocol);
+
+        [DllImport("libc", EntryPoint = "ioctl", CallingConvention = CallingConvention.Cdecl)]
+        private static extern int Ioctl3(int fd, uint request, ref ifreq ifr);
+
+        [DllImport("libc", EntryPoint = "bind", CallingConvention = CallingConvention.Cdecl)]
+        private static extern int BindSocket(int fd, ref CanSocketAddress addr, uint addrlen);
+
+        [DllImport("libc", EntryPoint = "close", CallingConvention = CallingConvention.Cdecl)]
+        private static extern int CloseSocket(int fd);
+
+        [DllImport("libc", EntryPoint = "write", CallingConvention = CallingConvention.Cdecl, SetLastError = true)]
```

Comment:
>SetLastError = true [](start = 94, length = 19)

why we set last error while I am not seeing you are using it?

Reply:
removing, error messaging is something we might want to improve in the future but I'd prefer to wait until this code can use Socket underneath instead of writing that logic right now (this will be possible once iot re-targets to 3.0)